packages/device_info/android/src/main/java/io/flutter/plugins/deviceinfo/DeviceInfoPlugin.java

```diff
@@ -4,120 +4,38 @@
+    plugin.setupMethodChannel(registrar.messenger(), registrar.context().getContentResolver());
   }
 
   @Override
-  public void onMethodCall(MethodCall call, Result result) {
-    if (call.method.equals("getAndroidDeviceInfo")) {
-      Map<String, Object> build = new HashMap<>();
-      build.put("board", Build.BOARD);
-      build.put("bootloader", Build.BOOTLOADER);
-      build.put("brand", Build.BRAND);
-      build.put("device", Build.DEVICE);
-      build.put("display", Build.DISPLAY);
-      build.put("fingerprint", Build.FINGERPRINT);
-      build.put("hardware", Build.HARDWARE);
-      build.put("host", Build.HOST);
-      build.put("id", Build.ID);
-      build.put("manufacturer", Build.MANUFACTURER);
-      build.put("model", Build.MODEL);
-      build.put("product", Build.PRODUCT);
-      if (VERSION.SDK_INT >= VERSION_CODES.LOLLIPOP) {
-        build.put("supported32BitAbis", Arrays.asList(Build.SUPPORTED_32_BIT_ABIS));
-        build.put("supported64BitAbis", Arrays.asList(Build.SUPPORTED_64_BIT_ABIS));
-        build.put("supportedAbis", Arrays.asList(Build.SUPPORTED_ABIS));
-      } else {
-        build.put("supported32BitAbis", Arrays.asList(EMPTY_STRING_LIST));
-        build.put("supported64BitAbis", Arrays.asList(EMPTY_STRING_LIST));
-        build.put("supportedAbis", Arrays.asList(EMPTY_STRING_LIST));
-      }
-      build.put("tags", Build.TAGS);
-      build.put("type", Build.TYPE);
-      build.put("isPhysicalDevice", !isEmulator());
-      build.put("androidId", getAndroidId());
-
-      Map<String, Object> version = new HashMap<>();
-      if (VERSION.SDK_INT >= VERSION_CODES.M) {
-        version.put("baseOS", VERSION.BASE_OS);
-        version.put("previewSdkInt", VERSION.PREVIEW_SDK_INT);
-        version.put("securityPatch", VERSION.SECURITY_PATCH);
-      }
-      version.put("codename", VERSION.CODENAME);
-      version.put("incremental", VERSION.INCREMENTAL);
-      version.put("release", VERSION.RELEASE);
-      version.put("sdkInt", VERSION.SDK_INT);
-      build.put("version", version);
-
-      result.success(build);
-    } else {
-      result.notImplemented();
-    }
+  public void onAttachedToEngine(FlutterPlugin.FlutterPluginBinding binding) {
+    setupMethodChannel(
+        binding.getFlutterEngine().getDartExecutor(),
+        binding.getApplicationContext().getContentResolver());
   }
 
-  /**
-   * Returns the Android hardware device ID that is unique between the device + user and app
-   * signing. This key will change if the app is uninstalled or its data is cleared. Device factory
-   * reset will also result in a value change.
-   *
-   * @return The android ID
-   */
-  @SuppressLint("HardwareIds")
-  private String getAndroidId() {
-    return Settings.Secure.getString(context.getContentResolver(), Settings.Secure.ANDROID_ID);
+  @Override
+  public void onDetachedFromEngine(FlutterPlugin.FlutterPluginBinding binding) {
+    channel.setMethodCallHandler(null);
```

Comment:
Recommendation: consider defining a `teardownMethodChannel()` for symmetry with `setupMethodChannel()`.